stl/inc/filesystem

```diff
@@ -1786,8 +1812,11 @@ namespace filesystem {
         static string _Pretty_message(const string_view _Op, const path& _Path1, const path& _Path2 = {}) {
             using namespace _STD string_view_literals; // TRANSITION, VSO-571749
             string _Result;
-            const string _Path1_str = _Path1.string();
-            const string _Path2_str = _Path2.string();
+            // Convert the paths to narrow encoding in a way that gracefully handles non-encodeable characters
+            const string _Path1_str = _Convert_wide_to_narrow_replace_chars<char_traits<char>>(
+                __std_fs_code_page(), _Path1.native(), allocator<char>());
+            const string _Path2_str = _Convert_wide_to_narrow_replace_chars<char_traits<char>>(
+                __std_fs_code_page(), _Path2.native(), allocator<char>());

```

Comment:
We've started following a convention of using empty braces to construct temporary objects; this makes code slightly more readable since they don't look like function calls.
Also, while the runtime performance doesn't really matter, there should be a very slight codegen size benefit to extracting the non-inlinable call to `__std_fs_code_page()`. (Note: haven't compiled or tested this, but I've brain-simulated clang-format)
```suggestion
            const auto _Code_page   = __std_fs_code_page();
            const string _Path1_str = _Convert_wide_to_narrow_replace_chars<char_traits<char>>(_Code_page,
                _Path1.native(), allocator<char>{});
            const string _Path2_str = _Convert_wide_to_narrow_replace_chars<char_traits<char>>(_Code_page,
                _Path2.native(), allocator<char>{});
```